### PR TITLE
OCM-10144 | fix: List upgrade for cluster asking machinepool identifier producing error

### PR DESCRIPTION
**Pull request by Mischulee**
https://issues.redhat.com/browse/OCM-10141

### CODE REVIEW

File: pkg/upgrade/upgrade.go [hunk 1]
```diff
@@ -21,6 +21,9 @@ func NewUpgradeArgsFunction(flagSource bool) func(cmd *cobra.Command, argv []str
 
 		if flagSource {
 			flags := cmd.Flags()
+			if !cmd.Flags().Changed(machinepoolFlagName) {
```
Comment (robpblake) [inline pkg/upgrade/upgrade.go:24]: @Mischulee Could you add to our existing unit test to verify that if the flag has not been changed, we do not perform the validation.

cheers!

Rob

Comment (Mischulee) [reply pkg/upgrade/upgrade.go:24]: Added!

**Review by robpblake — APPROVED** [review]
/lgtm

**Pull request merged**